actors/evm/src/interpreter/instructions/ext.rs

```diff
@@ -6,30 +14,74 @@ use {
 
 #[inline]
 pub fn extcodesize<'r, BS: Blockstore, RT: Runtime<BS>>(
-    _state: &mut ExecutionState,
-    _platform: &'r System<'r, BS, RT>,
+    state: &mut ExecutionState,
+    platform: &'r System<'r, BS, RT>,
 ) -> Result<(), StatusCode> {
-    // TODO
-    //  1. call actor::get_actor_code_cid
-    //  2. check that it matches our code CID (it's an EVM actor)
-    //  3. call GetEvmBytecode method, returns the CID of the EVM bytecode block
-    //  4. open the block
-    //  5. return the length
-    todo!()
+    let addr = state.stack.pop();
+    let len = get_evm_bytecode_cid(platform.rt, addr)
+        .and_then(|cid| get_evm_bytecode(platform.rt, &cid))
+        .map(|bytecode| bytecode.len())?;
+
+    state.stack.push(len.into());
+    Ok(())
 }
 
 pub fn extcodehash<'r, BS: Blockstore, RT: Runtime<BS>>(
-    _state: &mut ExecutionState,
-    _platform: &'r System<'r, BS, RT>,
+    state: &mut ExecutionState,
+    platform: &'r System<'r, BS, RT>,
 ) -> Result<(), StatusCode> {
-    // TODO
-
-    todo!();
+    let addr = state.stack.pop();
+    let cid = get_evm_bytecode_cid(platform.rt, addr)?;
+    // Take the first 32 bytes of the Multihash
+    state.stack.push(cid.hash().digest()[..32].into());
+    Ok(())
 }
 
 pub fn extcodecopy<'r, BS: Blockstore, RT: Runtime<BS>>(
-    _state: &mut ExecutionState,
-    _platform: &'r System<'r, BS, RT>,
+    state: &mut ExecutionState,
+    platform: &'r System<'r, BS, RT>,
 ) -> Result<(), StatusCode> {
-    todo!();
+    let ExecutionState { stack, .. } = state;
+    let (addr, dest_offset, data_offset, size) =
+        (stack.pop(), stack.pop(), stack.pop(), stack.pop());
+    let bytecode = get_evm_bytecode_cid(platform.rt, addr)
+        .and_then(|cid| get_evm_bytecode(platform.rt, &cid))?;
+
+    copy_to_memory(&mut state.memory, dest_offset, data_offset, size, bytecode.as_slice())?;
+
+    Ok(())
+}
+
+fn get_evm_bytecode_cid<BS: Blockstore, RT: Runtime<BS>>(
+    rt: &RT,
+    addr: U256,
+) -> Result<Cid, StatusCode> {
+    let addr =
+        Address::try_from(addr)?.as_id_address().expect("no support for non-ID addresses yet");
+
+    let evm_cid = rt.get_code_cid_for_type(Type::EVM);
+    let target_cid = rt.get_actor_code_cid(&addr.id().expect("not an ID address"));
+
+    if Some(evm_cid) != target_cid {
+        return Err(StatusCode::InternalError(
```

Comment:
that's probably an IllegalArgument and not an InternalError, it's a user error.

Reply:
I'll add an enum entry for `StatusCode::InvalidArgument` -- it's not an _illegal_ argument (it's still a well formed ID address), but an invalid one.